RFID.py

```diff
@@ -0,0 +1,148 @@
+            time.sleep(0.01)
+    return rfid_value
+
+
+def tts(text):
+    # Function to convert text to speech
+    engine.say(text)
+    engine.runAndWait()
+
+
+def authenticate_card(card_id):
+    # Function to authenticate the RFID card
+    if card_id == AUTHORIZED_CARD_ID0:
+        tts("Welcome, SentryCoder.Devloper")    #where SentryCoder.developer is meant define rfid card 0 for the owner of the robot change it for yourself
+        return True
+    if card_id == AUTHORIZED_CARD_ID1:
+        tts("Welcome,'name' ")
+        return True
+    if card_id == AUTHORIZED_CARD_ID2:
+        tts("Welcome,'name'")
+        return True
+    if card_id == AUTHORIZED_CARD_ID3:
+        tts("Welcome,'name'")
+        return True
+    if card_id == AUTHORIZED_CARD_ID4:
+        tts("Welcome,'name'")
+        return True
+    if card_id == AUTHORIZED_CARD_ID5:
+        tts("Welcome,'name'")
+        return True    
+    else:
+        tts("Unauthorized person")
+        return False
+
+def get_active_duration():
+    # Function to get the desired active duration from the user's voice input
+    tts("How long do you want SentryBOT to stay active?")
+    time.sleep(0.5)
+    tts('For example, say "I want SentryBOT to stay active for 1 hour."')
+    r = sr.Recognizer()
+    while True:
+        try:
+            with sr.Microphone() as source:
+                audio = r.listen(source)
+
+            text = r.recognize_google(audio, language="tr-TR")
+            duration = extract_duration_from_text(text)
+            if duration is not None:
+                break
+            else:
+                tts("Invalid duration, please try again.")
+        except (sr.UnknownValueError, sr.RequestError):
+            tts("Voice not understood, please try again.")
+
+    return duration
+
+
+def extract_duration_from_text(text):
+    # Function to extract duration from the text
+    match = re.search(r"(\d+)\s*(?:hour|hourly)", text)
+    if match:
+        duration = int(match.group(1))
+        return duration
+    else:
+        return None
+
+
+def open_jetson_session():
+    # Function to open Jetson Nano session
+    os.system("sudo -S <<< {} startx".format(ROOT_PASSWORD))
+
+
+def close_jetson_session():
+    # Function to close Jetson Nano session
+    os.system("sudo -S <<< {} pkill -u jetson".format(ROOT_PASSWORD))
+
+
+def main():
+    unauthorized_attempts = 0
+    wait_time = 0
+
+ 
+def main():
+    while True:
+        tts("Please authenticate your identity.")
+        card_id = read_rfid()
+        if authenticate_card(card_id):
+            active_duration = get_active_duration()
+            tts("SentryBOT activated for {} hours.".format(active_duration))
+            open_jetson_session()
+            time.sleep(active_duration * 60 * 60)
+            close_jetson_session()
+            tts("SentryBOT active duration completed. Wait for 10 seconds to authenticate your identity.")
+            time.sleep(10)
+
+
+if __name__ == "__main__":
```

Comment:
As discussed, this isn't a module so we can't use it without it being converted. But if you want to move it to the scripts folder it can work as a standalone script.